StringMatch/KMPStringMatcher.java

```diff
@@ -0,0 +1,76 @@
+package StringMatch;
```

Comment:
Kindly add a desciption to each solution for better understanding. 
It will be better if you add the StringMatch inside a String Algo folder.